File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    answer = ''
    part_count = Counter(participant)
    
    for comp in completion:
        part_count[comp] -= 1
    
    for person, cnt in part_count.items():
        if cnt == 1:
            return person
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> people;
    
    for(int i = 0; i<participant.size(); i++){
        if (people.find(participant[i]) != people.end()){
            people[participant[i]] += 1;
        }
        else{
            people[participant[i]] = 1;
        }
    }
    
    for(int j=0; j<completion.size(); j++){
        people[completion[j]] -= 1;
    }
    
    for(const auto& [person, cnt] : people){
        if (cnt == 1){
            answer = person;
            break;
        }
    }
    
    return answer;
}